packages/strapi-tiptap-editor/admin/src/utils/generateLabel.ts

```diff
@@ -0,0 +1,59 @@
+import truncate from 'lodash.truncate';
+import striptags from 'striptags';
+
+interface GenerateLabelData {
+  name: string;
+  content: string;
+  truncateLength?: number;
+}
+interface GenerateLabelReturnType {
+  label: string;
+  name: string;
+  labelKey: string;
+  content: string;
+}
+export const generateLabel = ({ name, content, truncateLength = 70 }: GenerateLabelData): GenerateLabelReturnType => {
+  if (!content?.trim()) return { label: '', name, labelKey: '', content: '' };
+
+  // Replace block elements with spaces to maintain text flow before stripping HTML tags
+  const formattedContent = content.replace(/<\/?(p|div|h[1-6]|li|blockquote|tr|td)(\s[^>]*)?\s*\/?>/gi, ' ');
+
+  // Decode HTML entities before stripping HTML tags
+  const decodedContent =
+    new DOMParser().parseFromString(formattedContent, 'text/html').documentElement.textContent || '';
```

Comment:
Dit is een goed idee, maar heb je dan nog de `replace` erboven nodig?

Nieuw idee: zou je hier niet DOMPurify voor kunnen gebruiken, en dan instellen dat je vrijwel niets mag, geen enkele tag?